pkg/util/filesystem/defaultfs.go

```diff
@@ -20,80 +20,102 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 	"time"
 )
 
 // DefaultFs implements Filesystem using same-named functions from "os" and "io/ioutil"
-type DefaultFs struct{}
+type DefaultFs struct {
+	root string
```

Comment:
if the root is empty, the behavior remains the same as before. we have a new method `NewTempFs` which injects a temporary directory as the root directory, all operations take this root into account.

Reply:
is filepath.Join(“”, …) a no-op, or does it prefix a /? is the behavior the same on windows if the arg paths are fully qualified? Is this util used on windows?

Reply:
answering my own question… Join ignores leading “” args. That said, the behavior is different (it passes the results through a Clean call which can normalize and is expensive), so I’d suggest a “prefix(path)” method that noops if root is zero-length

Reply:
on it!